exercises/Daniel_Murphy/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <functional>
+#include <iomanip>
+#include <sstream>
+
+std::vector<std::string> AnalyseLine(std::string line)
+{
+  // Remove punctuation that accompanies a space
+  std::vector<char> punctuation = {'?','.',',', '\'', '"', '(', ')', '!', ':', ';'};
+  for (char mark : punctuation)
+  {
+    line.erase(std::remove(line.begin(), line.end(),mark), line.end());
+  }
+
+  // Hyphens don't accompany spaces so replace with a space and split words
+  std::replace( line.begin(), line.end(), '-', ' ');
+
+  // force all letters to case lower
+  std::transform(line.begin(), line.end(), line.begin(), [](unsigned char c){ return std::tolower(c);});
+
+  std::vector<std::string> words;
+
+  std::stringstream lineOfText(line);
+  std::string word;
+  while (lineOfText >> word) {
+    if (word.size() > 4)
+      words.push_back(word);
+  }
+
+  return words;
+}
+
+std::map<std::string, int> UpdateMap(const std::vector<std::string> &words, std::map<std::string,int>& Counter)
+{
+  for (auto& word : words)
+  {
+    // if word already a member, increment its count by 1
+    Counter[word]++ ;
+  }
+  return Counter;
+}
+
+// https://www.geeksforgeeks.org/sorting-a-map-by-value-in-c-stl/
+bool cmp(const std::pair<std::string, int>& a, const std::pair<std::string, int>& b)
+{
+  return a.second > b.second;
+}
+
+// Convert map into vector of pairs, sorted by value of counts
+std::vector<std::pair<std::string, int> > sort_by_value(const std::map<std::string, int>& M) 
+{
+  // Declare vector of pairs 
+  std::vector<std::pair<std::string, int> > A; 
+
+  // Copy key-value pair from Map to vector of pairs 
+  for (auto& it : M) { A.push_back(it); } 
+
+  // Sort using comparator function 
+  sort(A.begin(), A.end(), cmp); 
```

Comment:
Although this compiles it's good practice to write std::sort rather than simply sort.
I think it compiles OK as it is because when the compiler doesn't find "sort" in the current namespace it spots that the parameters are defined in the std namespace so it goes off and looks for a sort in that namespace. If you defined your own sort though the absence of the prefix cause confusion over which one is being called